app/models/access.rb

```diff
@@ -0,0 +1,102 @@
+class Access < ApplicationRecord
+  ALLOWED_RESOURCE_TYPES = %w[Organization FacilityGroup Facility].freeze
+
+  belongs_to :user
+  belongs_to :resource, polymorphic: true, optional: true
+
+  enum role: {
+    super_admin: "super_admin",
+    admin: "admin",
+    analyst: "analyst"
+  }
+
+  validates :role, presence: true
+  validates :resource_type, inclusion: {in: ALLOWED_RESOURCE_TYPES}, allow_nil: true
+
+  class << self
+    def can_manage_facilities?(record)
+      return true if super_admin.exists?
+
+      manageable_facilities = facilities(:manage)
+
+      if record.instance_of?(Facility)
+        manageable_facilities.where(id: record).exists?
+      else
+        manageable_facilities.exists?
```

Comment:
Same as below, what else is a valid argument other than an instance of a Facility?  We don't need class level checks anymore, we have concrete objects we can work with.  Please correct me if I'm wrong, but I can't think of where you would do an auth check like this _without_ facilities to pass into the method.

If you want to ask if an admin can add a new Facility, you ask the parent object: `admin.can_manage?(facility_group)` -> if that returns true, the user can add a facility underneath the parent facility group.

Reply:
The only case is that you may want to check if you can manage _any_ facility. Similarly for _any_ Org or _any_ FacilityGroup – an [example](https://github.com/simpledotorg/simple-server/blob/master/app/controllers/admin/facilities_controller.rb#L12). 

Don't know of a good way to handle that case, other than passing in the whole object-level scope.

We could consider doing, no arguments for that case maybe, `admin.can_manage_facilities?`. wdyt?

Reply:
I think Rob's question here is - when do we need to check if you can manage _any_ facility? Do we have a use case for that today?

Reply:
Yeah, I posted an example in the comment above, a lot of `index` pages will be like that I think.